classes/controllers/FrmSimpleBlocksController.php

```diff
@@ -0,0 +1,207 @@
+			FrmAppHelper::plugin_url() . '/css/blocks.editor.build.css',
+			array( 'wp-edit-blocks' ),
+			$version
+		);
+	}
+
+	/**
+	 * Get the View type (show_count) for each View, e.g. calendar, dynamic
+	 *
+	 * @return array|object|void|null
+	 */
+	private static function get_show_counts() {
+		$show_counts = self::get_meta_values( 'frm_show_count', 'frm_display' );
+
+		return $show_counts;
+	}
+
+	/**
+	 * Get the specified meta value for the specified post type
+	 *
+	 * @param string $key
+	 * @param string $post_type
+	 *
+	 * @return array|object|void|null
+	 */
+	private static function get_meta_values( $key = '', $post_type = 'frm_display' ) {
+
+		global $wpdb;
+
+		if ( empty( $key ) ) {
+			return;
+		}
+
+		$results = $wpdb->get_results(
+			$wpdb->prepare(
+				"SELECT pm.post_id, pm.meta_value, pm.meta_key FROM {$wpdb->postmeta} pm LEFT JOIN {$wpdb->posts} p ON p.ID = pm.post_id WHERE ( pm.meta_key = %s ) AND p.post_type = %s",
+				$key,
+				$post_type
+			),
+			OBJECT_K
+		);
+
+		return $results;
+	}
+
+	/**
+	 * Get the options for the site's Views
+	 *
+	 * @return array|object|void|null
+	 */
+	private static function get_frm_options_for_views() {
+
+		$views_options = self::get_meta_values( 'frm_options', 'frm_display' );
+
+		foreach ( $views_options as $key => $value ) {
+			$views_options[ $key ]->meta_value = unserialize( $value->meta_value );
+		}
+
+		return $views_options;
+	}
+
+	/**
+	 * Registers simple form and View blocks
+	 *
+	 */
+	public static function register_guten_blocks() {
+		if ( ! is_callable( 'register_block_type' ) ) {
+			return;
+		}
+
+		register_block_type(
+			'formidable/simple-form',
+			array(
+				'attributes'      => array(
+					'form_id'     => array(
+						'type' => 'string',
+					),
+					'title'       => array(
+						'type' => 'string',
+					),
+					'description' => array(
+						'type' => 'string',
+					),
+					'minimize'    => array(
+						'type' => 'string',
+					),
+				),
+				'editor_script'   => 'formidable_simple-block-js',
+				'render_callback' => array( 'FrmSimpleBlocksController', 'simple_form_render' ),
+			)
+		);
+
+		register_block_type(
+			'formidable/simple-view',
+			array(
+				'attributes'      => array(
+					'view_id'           => array(
+						'type' => 'string',
+					),
+					'filter'            => array(
+						'type' => 'string',
+					),
+					'limit'             => array(
+						'type' => 'string',
+					),
+					'use_default_limit' => array(
+						'type'    => 'boolean',
+						'default' => false,
+					),
+				),
+				'editor_script'   => 'formidable_simple-block-js',
+				'render_callback' => array( 'FrmSimpleBlocksController', 'simple_view_render' ),
+			)
+		);
+	}
+
+	/**
+	 * Renders a form given the specified attributes.
+	 *
+	 * @param $attributes
+	 *
+	 * @return string
+	 */
+	public static function simple_form_render( $attributes ) {
+		$params = '';
+		$params .= self::create_attribute_text( 'id', $attributes['form_id'] );
```

Comment:
Would this be more clean if we keep it as an array?
```suggestion
		$params = array_filter( $attributes );
		$params['id'] = $params['form_id'];
```

Then load the form with FrmFormsController::get_form_shortcode( $params );

Reply:
An array definitely works nicely -- and looks better! --  with these limited options using ServerSideRender.

Reply:
I was in the mindset of using the API and having an extra params field, where it made more sense to me to use do_shortcode.